packages/flutter/lib/src/cupertino/refresh.dart

```diff
@@ -0,0 +1,520 @@
+/// completed and the indicator sliver has retracted at least 90% of the way
+/// back.
+///
+/// Can only be used in downward scrolling vertical lists.
+///
+/// See also:
+///
+///  * [CustomScrollView], a typical sliver holding scroll view this control
+///    should go into.
+///  * <https://developer.apple.com/ios/human-interface-guidelines/controls/refresh-content-controls/>
+///  * [RefreshIndicator], a Material Design version of the pull-to-refresh
+///    paradigm. This widget works differently than [RefreshIndicator] because
+///    instead of being an overlay on top of the scrollable, the
+///    [CupertinoRefreshControl] is part of the scrollable and actively occupies
+///    scrollable space.
+class CupertinoRefreshControl extends StatefulWidget {
+  /// Create a new [CupertinoRefreshControl] for inserting into a list of slivers.
+  ///
+  /// [refreshTriggerPullDistance], [refreshIndicatorExtent] both have reasonable
+  /// defaults and cannot be null.
+  ///
+  /// [builder] has a default indicator builder but can be null, in which case
+  /// no indicator UI will be shown but the [onRefresh] will still be invoked.
+  ///
+  /// [onRefresh] will be called when pulled far enough to trigger a refresh.
+  const CupertinoRefreshControl({
+    this.refreshTriggerPullDistance: _kDefaultRefreshTriggerPullDistance,
+    this.refreshIndicatorExtent: _kDefaultRefreshIndicatorExtent,
+    this.builder: buildSimpleRefreshIndicator,
+    this.onRefresh,
+  }) : assert(refreshTriggerPullDistance != null && refreshTriggerPullDistance > 0.0),
+       assert(refreshIndicatorExtent != null && refreshIndicatorExtent >= 0.0),
+       assert(
+         refreshTriggerPullDistance >= refreshIndicatorExtent,
+         'The refresh indicator cannot take more space in its final state '
+         'than the amount initially created by overscrolling'
+       );
+
+  /// The amount of overscroll the scrollable must be dragged to trigger a reload.
+  ///
+  /// Must not be null, must be larger than 0 and larger than [refreshIndicatorExtent].
+  ///
+  /// When overscrolled past this distance, [onRefresh] will be called if not
+  /// null and the [builder] will build in the [RefreshIndicatorMode.armed] state.
+  final double refreshTriggerPullDistance;
+
+  /// The amount of space the refresh indicator sliver will keep holding while
+  /// [onRefresh]'s [Future] is still running.
+  ///
+  /// Must not be null and must be positive. But can be 0, in which case the
+  /// sliver will start retracting back to 0 as soon as the refresh started.
+  ///
+  /// Must be smaller than [refreshTriggerPullDistance] since the sliver shouldn't
+  /// grow more after triggering the refresh.
+  final double refreshIndicatorExtent;
+
+  /// A builder that's called as this sliver's size changes and as the state
+  /// changes.
+  ///
+  /// A default simple Twitter-style pull-to-refresh indicator is provided if
```

Comment:
I'm not sure the "Twitter-style" adds anything (I, for one, have never seen Twitter's indicator, so I have no idea what it looks like).  I think just saying "A default pull-to-refresh indicator..." is fine, and doesn't endorse a product.

Reply:
It's not about twitter per se. The pull to refresh UI pattern is upstreamed into the iOS framework from the twitter app originally. It's just referring to the 'original' style pull to refresh as opposed to the iOS 11 new style which will be added in the next pr